src/clients/aggregator_client/api_types.rs

```diff
@@ -41,38 +46,26 @@ impl From<Vdaf> for VdafInstance {
     }
 }
 
-#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
-#[repr(u8)]
-pub enum Role {
-    Collector = 0,
-    Client = 1,
-    Leader = 2,
-    Helper = 3,
-}
-
-impl Role {
-    pub fn is_leader(&self) -> bool {
-        matches!(self, Self::Leader)
+impl TryFrom<task::HpkeConfig> for HpkeConfig {
+    type Error = Box<dyn std::error::Error + Send + Sync>;
+    fn try_from(value: task::HpkeConfig) -> Result<Self, Self::Error> {
+        Ok(Self::new(
+            value.id.unwrap().into(),
+            value.kem_id.unwrap().try_into()?,
+            value.kdf_id.unwrap().try_into()?,
+            value.aead_id.unwrap().try_into()?,
+            value.public_key.unwrap().into_bytes().into(),
+        ))
     }
 }
-
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct HpkeConfig {
-    pub id: u8,
-    pub kem_id: u8,
-    pub kdf_id: u8,
-    pub aead_id: u8,
-    pub public_key: Vec<u8>,
-}
-
-impl From<task::HpkeConfig> for HpkeConfig {
-    fn from(value: task::HpkeConfig) -> Self {
+impl From<HpkeConfig> for task::HpkeConfig {
+    fn from(hpke_config: HpkeConfig) -> Self {
         Self {
-            id: value.id.unwrap(),
-            kem_id: value.kem_id.unwrap(),
-            kdf_id: value.kdf_id.unwrap(),
-            aead_id: value.aead_id.unwrap(),
-            public_key: value.public_key.unwrap().into_bytes(),
+            id: Some((*hpke_config.id()).into()),
+            kem_id: Some((*hpke_config.kem_id()) as u16),
+            kdf_id: Some((*hpke_config.kdf_id()) as u16),
+            aead_id: Some((*hpke_config.aead_id()) as u16),
+            public_key: Some(URL_SAFE_NO_PAD.encode(hpke_config.public_key())),
```

Comment:
This duplicates https://github.com/divviup/janus/pull/1356. It seems preferable for the string representation to live in janus_messages and to just call ToString::to_string here